devtools/dev-requirements_conda.yml

```diff
@@ -15,9 +15,9 @@ dependencies:
   - pip:
     # Conda only parses a single list of pip requirements.
     # If two pip lists are given, all but the last list is skipped.
+    - jax >= 0.4.24, < 0.5.0
```

Comment:
I prefer to keep these files in alphabetical order

Reply:
See discussion here https://github.com/PlasmaControl/DESC/pull/1293#discussion_r1808063333. For general usage, you are right alphabetical order is better, but for `test_jax_versions` workflow having `jax` at the top and then installing the rest according to the `jax` version is helping `pip` a lot. Since many of our dependencies have `jax` dependency themselves, `pip` struggles with having the reverse order of installation.